### DISCUSSION TITLE
How to create a menu with conditional rendering

### DISCUSSION THREAD

**Discussion by nsmichelj** [ant-design/ant-design#40537] [Q&A]
Hello how are you, I need help to create a menu with conditional rendering, what I need is to show the menu options depending on the user role, I've been researching on the net but I couldn't find a solution, I hope someone can help me... thank you.

**Comment by gustavoabel** [comment]
Here's an example of how you can create a menu with conditional rendering using Ant Design and ReactJS:

```
import { Menu } from 'antd';

const { SubMenu } = Menu;

const UserMenu = ({user}) => {
  return (
    <Menu>
      {user.isAdmin && (
        <SubMenu title="Admin">
          <Menu.Item>Option 1</Menu.Item>
          <Menu.Item>Option 2</Menu.Item>
        </SubMenu>
      )}
      {user.isEditor && (
        <SubMenu title="Editor">
          <Menu.Item>Option 3</Menu.Item>
          <Menu.Item>Option 4</Menu.Item>
        </SubMenu>
      )}
      <SubMenu title="General">
        <Menu.Item>Option 5</Menu.Item>
        <Menu.Item>Option 6</Menu.Item>
      </SubMenu>
    </Menu>
  );
};
```

This example checks the user.isAdmin and user.isEditor properties and shows the relevant menu options. If the user is not an admin or an editor, then they will only see the "General" options.

**Reply by gustavoabel** [reply]
I hope I can help you :)

**Reply by nsmichelj** [reply]
creating the menu like that was not deprecated?

**Reply by gustavoabel** [reply]
I don't think so, I create it this way

**Reply by nsmichelj** [reply]
I was finally able to try your example, and indeed, that way of doing it is deprecated according to these warnings in the console:

![Captura](https://user-images.githubusercontent.com/76160336/218878813-5d5673e1-b2cf-4537-abd2-4072e588fb62.PNG)

I don't know exactly from which version but I'm using 5.1.
